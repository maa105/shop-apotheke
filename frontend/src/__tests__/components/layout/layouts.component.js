import React from 'react';
import { render } from '@testing-library/react';

import LayoutComponent from '../../../components/layout/layout.component';
import { Layout } from 'antd';

jest.mock('../../../components/layout/top/top.component', () => {
  return jest.fn(() => <div>Top Mock</div>)
});

jest.mock('antd', () => {
  return {
    setState: (_state) => { state = _state; },
    Layout: {
      Content: jest.fn(() => <div>Content Mock</div>),
      Header: jest.fn(() => <div>Header Mock</div>)
    }
  }
});

beforeEach(() => {
  jest.clearAllMocks();
});

it('layout renders header and content', () => {
  const component = render(
    <LayoutComponent>Content Text</LayoutComponent>
  );
  expect(component.asFragment()).toMatchSnapshot();
  
  expect(Layout.Content.mock.calls.length).toEqual(1);
  expect(Layout.Header.mock.calls.length).toEqual(1);

  const TopComponent = Layout.Header.mock.calls[0][0].children;
  const topComponent = render(
    TopComponent
  );
  expect(topComponent.asFragment()).toMatchSnapshot();

  const contentChildren = Layout.Content.mock.calls[0][0].children;
  expect(contentChildren).toEqual('Content Text');

});
