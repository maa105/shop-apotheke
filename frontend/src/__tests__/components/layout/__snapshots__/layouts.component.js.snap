// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout renders header and content 1`] = `
<DocumentFragment>
  <div>
    Header Mock
  </div>
  <div>
    Content Mock
  </div>
</DocumentFragment>
`;

exports[`layout renders header and content 2`] = `
<DocumentFragment>
  <div>
    Top Mock
  </div>
</DocumentFragment>
`;
